products/bash/setup-demos.sh

```diff
@@ -312,30 +347,110 @@ for DEMO in $(echo $REQUIRED_DEMOS_JSON | jq -r 'to_entries[] | select( .value.e
   done
 done
 
+#-------------------------------------------------------------------------------------------------------------------
+# Change status phase to Pending as installation starting
+#-------------------------------------------------------------------------------------------------------------------
+$DEBUG && divider && echo -e "$info [INFO] Changing the status to 'Pending' as installation is starting..."
+update_phase "Pending"
+
+#-------------------------------------------------------------------------------------------------------------------
+# Check if the namespace  and the secret exists
+#-------------------------------------------------------------------------------------------------------------------
+
+$DEBUG && divider && echo -e "$info [INFO] Check if the '$NAMESPACE' namespace and the secret 'ibm-entitlement-key' exists...\n"
+
+# add namespace to status if exists
+oc get project $NAMESPACE 2>&1 >/dev/null
+if [ $? -ne 0 ]; then
+  $DEBUG && echo -e "\n$cross [ERROR] Namespace '$NAMESPACE' does not exist" && divider
+  update_conditions "Namespace '$NAMESPACE' does not exist" "Getting"
+  update_phase "Failed"
+else
+  $DEBUG && echo -e "$tick [SUCCESS] Namespace '$NAMESPACE' already exists" && divider
+  NAMESPACE_TO_ADD=$(echo $NAMESPACE_OBJECT | jq -r '.name="'$NAMESPACE'" ')
+  STATUS=$(echo $STATUS | jq -c '.namespaces += ['"${NAMESPACE_TO_ADD}"']')
+fi
+
+# check if the secret exists in the namespace
+oc get secret -n $NAMESPACE ibm-entitlement-key 2>&1 >/dev/null
+if [ $? -ne 0 ]; then
+  $DEBUG && echo -e "\n$cross [ERROR] Secret 'ibm-entitlement-key' not found in '$NAMESPACE' namespace" && divider
+  update_conditions "Secret 'ibm-entitlement-key' not found in '$NAMESPACE' namespace" "Getting"
+  update_phase "Failed"
+fi
+
 #-------------------------------------------------------------------------------------------------------------------
 # Add the required addons
 #-------------------------------------------------------------------------------------------------------------------
-$DEBUG && divider
-$DEBUG && echo "Addons:"
+$DEBUG && echo -e "$info [INFO] Installing and setting up addons:"
 for row in $(echo "${REQUIRED_ADDONS_JSON}" | jq -r '.[] | select(.enabled == true ) | @base64'); do
+  divider
   ADDON_JSON=$(echo ${row} | base64 --decode)
-  $DEBUG && echo $ADDON_JSON | jq .
+  $DEBUG && echo ${ADDON_JSON} | jq . && echo ""
+  ADDON_TYPE=$(echo ${ADDON_JSON} | jq -r '.type')
+  case ${ADDON_TYPE} in
+  postgres)
+    echo -e "$info [INFO] Releasing postgres...\n"
+    if ! $SCRIPT_DIR/release-psql.sh -n $NAMESPACE; then
+      $DEBUG && echo -e "\n$cross [ERROR] Failed to release PostgreSQL in the '$NAMESPACE' namespace"
+      update_conditions "Failed to release PostgreSQL in the '$NAMESPACE' namespace" "Releasing"
```

Comment:
I think ERRORs should always be output, not just when DEBUG is true. Also, maybe change update_conditions to always output the message as an error (even when DEBUG is false), then don't need a seperate echo.

Reply:
Done